client/app/bundles/course/assessment/containers/AssessmentForm.jsx

```diff
@@ -62,14 +67,51 @@ const validate = (values) => {
 
 class AssessmentForm extends React.Component {
   static propTypes = {
+    dispatch: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func.isRequired,
+    start_at: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.instanceOf(Date),
+    ]),
+    end_at: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.instanceOf(Date),
+    ]),
     autograded: PropTypes.bool,
     password_protected: PropTypes.bool,
     submitting: PropTypes.bool,
     // Above are props from redux-form.
+
     onSubmit: PropTypes.func.isRequired,
+    // If the Form is in editing mode, `published` button will be displayed in editting mode.
+    editing: PropTypes.bool,
+    // If allow to switch between autoraded and manually graded mode.
+    modeSwitching: PropTypes.bool,
+    folderAttributes: PropTypes.shape({
+      folderId: PropTypes.number,
+      // See MaterialFormContainer for detail PropTypes.
+      materials: PropTypes.array,
+    }),
+    // Condtions will be displayed if the attributes are present.
+    conditionAttributes: PropTypes.shape({
+      new_condition_urls: PropTypes.array,
+      conditions: PropTypes.array,
+    }),
   };
 
+  onStartAtChange = (_, newStartAt) => {
```

Comment:
not very sure about the bonus_end, if we do that shall we shift bonus_end when end_at changes?